### DISCUSSION TITLE
Levee Implementation

### DISCUSSION THREAD

**Discussion by rnugent3** [HydrologicEngineeringCenter/fda-model#90] [General]
We have run into questions about the way that we should implement levees a couple times, it seems that we are not yet settled. 

For reference, see this issue: https://github.com/HydrologicEngineeringCenter/fda-model/issues/47#issuecomment-933807161 and this pull request: https://github.com/HydrologicEngineeringCenter/fda-model/pull/85

It seems that the questions are:

**1. How do we incorporate a fragility function into the compute of EAD?**
- It appears that we have answered this outside of the above referenced issue and PR. Stage damage is multiplied by the fragility function, reducing damage for stages below the top of levee elevation (but not to zero as without the fragility function), damage for stages at or above the top of levee elevation are unchanged (i.e. 100%*damage = damage). 
- This is consistent with the implementation in HEC-FDA 1.4.x. 

**2. At what elevation is the probability of levee failure 100%? Related, what does the probability of failure represent?**
- The probability of failure represents the probability that water from the channel gets into the floodplain - _not how much water gets into the floodplain_, just whether or not water gets into the floodplain. How much water gets into the floodplain is defined by the exterior-interior relationship. 
- HEC-FDA 1.4.x sets the probability of levee failure at the top elevation of the levee. 
- PB 2019-04 names overtopping without breach as an inundation scenario (water goes into the floodplain) 

**3. How do we handle overtopping?**
- Without an exterior-interior relationship, HEC-FDA 1.4.x equates the water level in the flood plain to the water level in the channel the moment that the water level reaches the top of levee elevation. The floodplain is completely inundated. The rate of overtopping as part of levee design is not evaluated in the EAD compute without an exterior-interior relationship. 
- With an exterior-interior relationship, overtopping without breach - part of levee design - is evaluated in the compute of EAD in HEC-FDA 1.4.x.

**Comment by rnugent3** [comment]
My preference is that we use the HEC-FDA Version 1 methodology in Version 2.0 as much as practicable, and leave refinement of the methodology for later versions.

**Comment by rnugent3** [comment]
Life Discussion between @rnugent3 @HenryGeorgist 17 Dec 21:

If 70% at 1ft below top of levee, set 100% at top of levee, interpolate. 
If 100% above top of levee, allow, but use top of levee in performance. But give warning that this way is not recommended because of PB 2019-04, and recommend to at a minimum include interior/exterior. 

the depth of overtopping above levee can lead to other types of breach 
interior/exterior definitely needed

**Comment by HenryGeorgist** [comment]
technically i think we accomplish this by adding a top of levee value in the builder of the with levee argument for the simulation class. We store the top of levee as a value along with the fragility curve. We would need to add some logic either as part of the curve (for uncertainty sampling might be included in the fragility curves) or as a preprocess to do the linear interpolation if the percentage of the fragility curve does not go to 100%.
